app/src/sm_at_socket.c

```diff
@@ -49,6 +49,12 @@ enum sm_socket_role {
 	AT_SOCKET_ROLE_SERVER
 };
 
+/**@brief Socket send modes. */
+enum sm_socket_send_mode {
+	AT_SOCKET_SEND_STRING,	/* String data in command */
+	AT_SOCKET_SEND_DATAMODE	/* Enter data mode */
```

Comment:
I would have the mode in the values too and the prefix similar to the enum name:
```suggestion
	SM_SOCKET_SEND_MODE_STRING,	/* String data in command */
	SM_SOCKET_SEND_MODE_DATA	/* Enter data mode */
```

@MarkusLassila opinions before this is changed so that it doesn't need to be done several times :smile:

Reply:
LGTM